lib/routemaster/middleware/siphon.rb

```diff
@@ -0,0 +1,28 @@
+module Routemaster
+  module Middleware
+    class Siphon
+      # Filters out events based on their topic and passes them to a handling class
+      # Usage:
+      #    use Middleware::Siphon, 'some_topic' => SomeTopicSeriesBuilder
+      def initialize(app, processors)
+        @app = app
+        @processors = processors
+      end
+
+      def topics_to_avoid
```

Comment:
This method should probably be `private`. Also might want to memoise the value as I'm not sure if `keys` allocates a new array each time.